packages/@aws-cdk/aws-apigateway/lib/integration.ts

```diff
@@ -0,0 +1,260 @@
+     *
+     * @default Caller identity is not passed through
+     */
+    credentialsPassthrough?: boolean;
+
+    /**
+     * Specifies the pass-through behavior for incoming requests based on the
+     * Content-Type header in the request, and the available mapping templates
+     * specified as the requestTemplates property on the Integration resource.
+     * There are three valid values: WHEN_NO_MATCH, WHEN_NO_TEMPLATES, and
+     * NEVER.
+     */
+    passthroughBehavior?: PassthroughBehavior
+
+    /**
+     * The request parameters that API Gateway sends with the backend request.
+     * Specify request parameters as key-value pairs (string-to-string
+     * mappings), with a destination as the key and a source as the value.
+     *
+     * Specify the destination by using the following pattern
+     * integration.request.location.name, where location is querystring, path,
+     * or header, and name is a valid, unique parameter name.
+     *
+     * The source must be an existing method request parameter or a static
+     * value. You must enclose static values in single quotation marks and
+     * pre-encode these values based on their destination in the request.
+     */
+    requestParameters?: { [dest: string]: string };
+
+    /**
+     * A map of Apache Velocity templates that are applied on the request
+     * payload. The template that API Gateway uses is based on the value of the
+     * Content-Type header that's sent by the client. The content type value is
+     * the key, and the template is the value (specified as a string), such as
+     * the following snippet:
+     *
+     *     { "application/json": "{\n    \"statusCode\": \"200\"\n}" }
+     *
+     * @see http://docs.aws.amazon.com/apigateway/latest/developerguide/api-gateway-mapping-template-reference.html
+     */
+    requestTemplates?: { [contentType: string]: string };
+
+    /**
+     * The response that API Gateway provides after a method's backend completes
+     * processing a request. API Gateway intercepts the response from the
+     * backend so that you can control how API Gateway surfaces backend
+     * responses. For example, you can map the backend status codes to codes
+     * that you define.
+     */
+    integrationResponses?: IntegrationResponse[];
+
+    /**
+     * The templates that are used to transform the integration response body.
+     * Specify templates as key-value pairs (string-to-string mappings), with a
+     * content type as the key and a template as the value.
+     *
+     * @see http://docs.aws.amazon.com/apigateway/latest/developerguide/api-gateway-mapping-template-reference.html
+     */
+    selectionPattern?: string;
+}
+
+export interface IntegrationProps {
+    /**
+     * Specifies an API method integration type.
+     */
+    type: IntegrationType;
+
+    /**
+     * The Uniform Resource Identifier (URI) for the integration.
+     *
+     * - If you specify MOCK for the `type` property, this is not required.
```

Comment:
Two lines later, you say `If you specify MOCK for the type property, don't specify this property.`. Only one of them is the truth, and I don't reckon it's this one.

Reply:
👍